src/kernel-install/ukify.py

```diff
@@ -0,0 +1,233 @@
+#!/usr/bin/python
+# SPDX-License-Identifier: LGPL-2.1+
+
+import argparse
+import dataclasses
+import itertools
+import pathlib
+import os
+import shlex
+import subprocess
+import tempfile
+import typing
+
+efi_arches = {
+        # host_arch: efi_arch
+        'x86'    : 'ia32',
+        'x86_64' : 'x64',
+        'arm'    : 'arm',
+        'aarch64': 'aa64',
+        'riscv64': 'riscv64',
+}
+
+def guess_efi_arch():
+    arch = os.uname().machine
+    efi_arch = efi_arches[arch]
+
+    if arch == 'x86_64':
+        try:
+            size = open('/sys/firmware/efi/fw_platform_size').read().strip()
```

Comment:
build machine might be different than the target machine? there should be at least an option to manually override

Reply:
This is only looked at if `--efi-arch` wasn't passed.

Reply:
The `--efi-arch` does this override. This function is only as a fallback.